motr/m0crate/crate_io.c

```diff
@@ -717,6 +717,12 @@ int cr_task_execute(struct m0_task_io *cti)
 			if (rc == 0)
 				cr_op_namei(cwi, cti, CR_DELETE);
 			break;
+		case CR_READ_ONLY:
+			rc = cr_op_namei(cwi, cti, CR_OPEN);
+			if (rc == 0) {
+				cr_op_io(cwi, cti, CR_READ);
+			}
+			break;
```

Comment:
What about the delete, dont we need delete operation here

Reply:
What if read_only was called without write

Reply:
> What about the delete, dont we need delete operation here

This opcode is added so that we can read exiting populated data and thus can be done multiple times explicitly or  with NR_ROUND option in workload. To deleted this data CLEANUP opcode can be used.

Reply:
> What if read_only was called without write

I did not checked this, but it should return ENOENT error if the objects itself are not present. Will check this.

Reply:
@yatin-mahajan, Please update here if you have checked and verified.